lib/src/deploy.rs

```diff
@@ -71,8 +103,14 @@ impl ImageState {
         &self,
         repo: &ostree::Repo,
     ) -> Result<Option<ostree_ext::oci_spec::image::ImageManifest>> {
-        ostree_container::store::query_image_commit(repo, &self.ostree_commit)
-            .map(|v| Some(v.manifest))
+        match self.backend {
+            Backend::OstreeContainer => {
+                ostree_container::store::query_image_commit(repo, &self.ostree_commit)
+                    .map(|v| Some(v.manifest))
+            }
+            // TODO: Figure out if we can get the OCI manifest from podman
```

Comment:
I don't think it is possible to fetch a manifest with a `podman` command; you'd need to fall back to `skopeo inspect` or something else

Reply:
We can continue to use https://github.com/containers/containers-image-proxy-rs/ for this stuff, see [this API](https://docs.rs/containers-image-proxy/latest/containers_image_proxy/struct.ImageProxy.html#method.fetch_manifest) which is what's used by the ostree-container code today.

We can consider extending the proxy API too with other things we need, as it's a reliable and tested bridge between the Rust and Go sides.

Reply:
Actually also, we could fork `skopeo copy` and not `podman pull` too.  That may help keep things clearer even in the short term.